stellargraph/mapper/sequences.py

```diff
@@ -530,6 +533,126 @@ def __getitem__(self, index):
+                    "expected the number of target values and the number of graphs to be the same length,"
+                    f"found {len(graphs)} graphs and {len(targets)} targets."
+                )
+
+            self.targets = np.asanyarray(targets)
+
+        if self.normalize_adj:
+            self.normalized_adjs = [
+                normalize_adj(graph.to_adjacency_matrix()) for graph in graphs
+            ]
+        else:
+            self.normalize_adjs = [graph.to_adjacency_matrix() for graph in graphs]
+
+        self.normalized_adjs = np.asanyarray(self.normalized_adjs)
+
+        self.on_epoch_end()
+
+    def __len__(self):
+        return int(np.ceil(len(self.graphs) / self.batch_size))
+
+    def __getitem__(self, index):
+        graphs = self.graphs[
+            index * self.batch_size : (index * self.batch_size) + self.batch_size
+        ]
+
+        adj_graphs = self.normalized_adjs[
+            index * self.batch_size : (index * self.batch_size) + self.batch_size
+        ]
+
+        # The number of nodes for the largest graph in the batch. We are going to pad with 0 rows and columns
+        # the adjacency and node feature matrices (only the rows in this case) to equal in size the adjacency and
+        # feature matrices of the largest graph.
+        max_nodes = max([graph.number_of_nodes() for graph in graphs])
+
+        graph_targets = None
+        if self.targets is not None:
+            graph_targets = self.targets[
+                index * self.batch_size : (index * self.batch_size) + self.batch_size
+            ]
+
+        # pad adjacency and feature matrices to equal the size of those from the largest graph
+        features = [
+            np.pad(
+                graph.node_features(graph.nodes()),
+                pad_width=((0, max_nodes - graph.number_of_nodes()), (0, 0)),
+            )
+            for graph in graphs
+        ]
+        features = np.stack(features)
+
+        for adj in adj_graphs:
+            adj.resize((max_nodes, max_nodes))
+        adj_graphs = np.stack([adj.toarray() for adj in adj_graphs])
+
+        masks = np.full((len(graphs), max_nodes), fill_value=False, dtype=np.bool)
+        for index, graph in enumerate(graphs):
+            masks[index, : graph.number_of_nodes()] = True
+
+        # features is array of dimensionality
+        #      batch size x max number of nodes in graph x node feature dimensionality
+        # masks is array of dimensionality
+        #      batch size x max number of nodes
+        # adj_graphs is array of dimensionality
+        #      batch size x max number of nodes x max number of nodes
+        # graph_targets is array of dimensionality
+        #      batch size x number of classes
+        return [features, masks, adj_graphs], graph_targets
+
+    def on_epoch_end(self):
+        """
+         Shuffle all graphs at the end of each epoch
+        """
+        indexes = list(range(len(self.graphs)))
```

Comment:
No objection about the suggested change, but I would prefer to tackle it as part of #1043 (see https://github.com/stellargraph/stellargraph/issues/1043#issuecomment-599840069) since this bit of code will very likely have to be updated for that ticket.